cypress/support/cypress-commands/assertion.js

```diff
@@ -18,120 +18,72 @@
-          resolve(contentObject);
-        } else {
-          reject(`Unable to find Error content validation object for ${errorType}`);
-        }
-      });
-    };
-
-    if (expectedContent === CONSTANTS.EXCEPTION_ERROR_OBJECT) {
-      cy.wrap(fetchErrorValidationObjectForExceptionmethod(method, param)).then((errorObject) => {
-        validateErrorObjects(errorObject);
-      });
+    //  If error object having type as "schemaOnly", skipping the content vallidation
+    if (expectedContent.type === CONSTANTS.SCHEMA_ONLY) {
+      return;
+    } // Doing custom validation when error object having type as "custom"
+    else if (expectedContent.type === CONSTANTS.CUSTOM) {
+      const additionalParams = { apiObject: apiObject, expectedContent: expectedContent };
+      cy.customValidation(expectedContent, additionalParams);
     } else {
       validateErrorObjects(expectedContent);
     }
 
     // Function to do content validation for the error objects.
-    function validateErrorObjects(errorSchemaObject) {
-      console.log(`Debug : errorSchemaObject ` + JSON.stringify(errorSchemaObject));
+    function validateErrorObjects(errorContentObject) {
+      console.log(`Debug : errorContentObject ` + JSON.stringify(errorContentObject));
       try {
+        // errorContentObject having expected format doing content validation else failing the test
         if (
-          errorSchemaObject &&
-          errorSchemaObject.hasOwnProperty('validations') &&
-          Array.isArray(errorSchemaObject.validations)
+          errorContentObject &&
+          errorContentObject.hasOwnProperty('validations') &&
+          Array.isArray(errorContentObject.validations)
         ) {
-          errorSchemaObject.validations.forEach((errorContentObject) => {
+          errorContentObject.validations.forEach((errorContentObject) => {
             errorContentObject = errorContentObject.type;
-
-            const apiOrEventObject = UTILS.getApiOrEventObjectFromGlobalList(
-              method,
-              context,
-              appId,
-              validationType
-            );
-            const apiErrorResponse =
-              validationType == CONSTANTS.EVENT
-                ? apiOrEventObject.eventListenerResponse.error
-                : apiOrEventObject.apiResponse.error;
-
-            fireLog.include(
-              errorContentObject.errorCode,
-              apiErrorResponse.code,
-              CONSTANTS.ERROR_CODE
-            );
+            fireLog.include(errorContentObject.errorCode, errorResponse.code, CONSTANTS.ERROR_CODE);
           });
         } else {
-          fireLog.fail(`Unable to find Error content validation object for ${errorSchemaObject}`);
+          fireLog.fail(`Unable to find Error content validation object for ${errorContentObject}`);
         }
       } catch (error) {
         fireLog.fail(error.message);
       }
     }
+
+    // Function to fetch the error object key name based on exception method.
```

Comment:
"fetchErrorValidationObjectForExceptionMethod" will fetch the error content object key name based on the exception list. 
Ex: If Authentication.token present in "NOT_SUPPORTED_METHODS" exception list, fetching the NOT_SUPPORTED key name from the error content json

Reply:
Can we add that info as js doc to the function?